src/GafferOSL/ShadingEngine.cpp

```diff
@@ -257,6 +259,88 @@ bool convertValue( void *dst, TypeDesc dstType, const void *src, TypeDesc srcTyp
 	return false;
 }
 
+struct ReplaceFunctor
```

Comment:
I'd be tempted to use a lambda for this these days, to remove the constructor and member data boilerplate.

Reply:
I'm pretty sure I find it more valuable that this exactly match the code it has been copied from, so that it's obvious to future maintainers that the two are the same?  Do you want me to update the Cortex one to use a lambda?